src/contractor/contractor.cpp

```diff
@@ -967,12 +944,12 @@ void Contractor::WriteCoreNodeMarker(std::vector<bool> &&in_is_core_node) const
         unpacked_bool_flags[i] = is_core_node[i] ? 1 : 0;
     }
 
-    boost::filesystem::ofstream core_marker_output_stream(config.core_output_path,
-                                                          std::ios::binary);
-    unsigned size = unpacked_bool_flags.size();
-    core_marker_output_stream.write((char *)&size, sizeof(unsigned));
-    core_marker_output_stream.write((char *)unpacked_bool_flags.data(),
-                                    sizeof(char) * unpacked_bool_flags.size());
+    storage::io::FileWriter core_marker_output_file(config.core_output_path,
+                                                    storage::io::FileWriter::HasNoFingerprint);
+
+    const auto count = unpacked_bool_flags.size();
+    core_marker_output_file.WriteElementCount32(count);
```

Comment:
Hm count should be a size_t and here we write only a four byte count - as you said, all of these should be eight bytes